bin/node-template/runtime/src/lib.rs

```diff
@@ -412,3 +413,19 @@ impl_runtime_apis! {
 		}
 	}
 }
+
+impl SignedExtensionProvider for Runtime {
```

Comment:
Why do we need this in `no_std`?

Reply:
that's because it's a subtrait of `frame_system::Trait`

Reply:
or is it super-trait?

```rust
pub trait SignedExtensionProvider: frame_system::Trait {
    /// Concrete SignedExtension type.
    type Extra: SignedExtension;

    /// build extras for inclusion in extrinsics
    fn construct_extras(nonce: IndexFor<Self>) -> Self::Extra;
}
```

Reply:
`frame_system::Trait` is the super trait of `SignedExtensionProvider`

Reply:
sub trait then.